pkg/solidity-utils/test/foundry/StableMath.t.sol

```diff
@@ -368,4 +371,96 @@ contract StableMathTest is Test {
 
         assertLe(invariantRatioDown, invariantRatioRegular, "Invariant ratio should have gone down");
     }
+
+    function testComputeInvariantLessThenInvariantWithLargeDelta__Fuzz(
+        uint256 amp,
+        uint256 tokenCount,
+        uint256 deltaCount,
+        uint256[8] memory indexes,
+        uint256[8] memory deltas,
+        uint256[8] memory balancesRaw
+    ) public view {
+        _testComputeInvariantLessThenInvariantWithDelta(
+            amp,
+            tokenCount,
+            deltaCount,
+            indexes,
+            deltas,
+            balancesRaw,
+            type(uint128).max
+        );
+    }
+
+    function testComputeInvariantLessThenInvariantWithSmallDelta__Fuzz(
+        uint256 amp,
+        uint256 tokenCount,
+        uint256 deltaCount,
+        uint256[8] memory indexes,
+        uint256[8] memory deltas,
+        uint256[8] memory balancesRaw
+    ) public view {
+        _testComputeInvariantLessThenInvariantWithDelta(
+            amp,
+            tokenCount,
+            deltaCount,
+            indexes,
+            deltas,
+            balancesRaw,
+            1000
+        );
+    }
+
+    function _testComputeInvariantLessThenInvariantWithDelta(
+        uint256 amp,
+        uint256 tokenCount,
+        uint256 deltaCount,
+        uint256[8] memory indexes,
+        uint256[8] memory deltas,
+        uint256[8] memory balancesRaw,
+        uint256 maxDelta
+    ) internal view {
+        amp = boundAmp(amp);
+        tokenCount = bound(tokenCount, MIN_TOKENS, MAX_TOKENS);
+        deltaCount = bound(deltaCount, 1, tokenCount);
+
+        uint256[] memory balances = new uint256[](tokenCount);
+        for (uint256 i = 0; i < tokenCount; i++) {
+            balances[i] = bound(balancesRaw[i], 1, type(uint128).max);
+        }
+
+        uint256[] memory newBalances = new uint256[](tokenCount);
+        ScalingHelpers.copyToArray(balances, newBalances);
+
+        for (uint256 i = 0; i < deltaCount; i++) {
+            uint256 tokenIndex = bound(indexes[i], 0, tokenCount - 1);
+            uint256 delta;
+            if (maxDelta > type(uint128).max - newBalances[tokenIndex]) {
```

Comment:
Consider locally defining `uint256 constant MAX_UINT128 = type(uint128).max` for readability/consistency. Probably not worth deriving from BaseTest vs. Test to inherit it, which would require overriding setup, etc.